.github/workflows/release.yaml

```diff
@@ -72,28 +72,21 @@ jobs:
         run: |
           brew tap mitchellh/gon
           brew install mitchellh/gon/gon
-      - name: build mirrord-layer x86-64
-        run: cargo +nightly-2024-09-12 build --release -p mirrord-layer --target=x86_64-apple-darwin
-      - name: build mirrord-layer macOS arm/arm64e
-        # Editing the arm64 binary, since arm64e can be loaded into both arm64 & arm64e
-        # >> target/debug/libmirrord_layer.dylib: Mach-O 64-bit dynamically linked shared library arm64
-        # >> magic bits: 0000000 facf feed 000c 0100 0000 0000 0006 0000
-        # >> After editing using dd -
-        # >> magic bits: 0000000 facf feed 000c 0100 0002 0000 0006 0000
-        # >> target/debug/libmirrord_layer.dylib: Mach-O 64-bit dynamically linked shared library arm64e
+        # Instead of spoofing an arm64e by changing bits, we now use a code shim compiled as arm64e that loads the arm64 layer 
+      - name: build mirrord-layer x86-64, arm and code shim
         run: |
           cargo +nightly-2024-09-12 build --release -p mirrord-layer --target=aarch64-apple-darwin
-          cp target/aarch64-apple-darwin/release/libmirrord_layer.dylib target/aarch64-apple-darwin/release/libmirrord_layer_arm64e.dylib
-          printf '\x02' | dd of=target/aarch64-apple-darwin/release/libmirrord_layer_arm64e.dylib bs=1 seek=8 count=1 conv=notrunc
-      - name: Sign layer binaries
+          cargo +nightly-2024-09-12 build --release -p mirrord-layer --target=x86_64-apple-darwin
+          clang -arch arm64e -dynamiclib -o target/aarch64-apple-darwin/release/shim.dylib scripts/shim.c
+      - name: Sign layer binaries and code shim
         env:
           AC_USERNAME: ${{ secrets.APPLE_DEVELOPER }}
           AC_PASSWORD: ${{ secrets.APPLE_DEVELOPER_PASSWORD }}
         run: gon -log-level=debug -log-json .github/workflows/arch_dylib_sign.json
       - name: create universal target dir
         run: mkdir -p /tmp/target/universal-apple-darwin/release
       - name: create universal dylib
-        run: lipo -create -output /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib target/aarch64-apple-darwin/release/libmirrord_layer.dylib target/x86_64-apple-darwin/release/libmirrord_layer.dylib target/aarch64-apple-darwin/release/libmirrord_layer_arm64e.dylib
+        run: lipo -create -output /tmp/target/universal-apple-darwin/release/libmirrord_layer.dylib target/aarch64-apple-darwin/release/shim.dylib target/x86_64-apple-darwin/release/libmirrord_layer.dylib
```

Comment:
Are you sure macOS will always just load the arm64e library if there is no arm64 library in the fat binary? Otherwise we could include the arm64 binary in the fat binary.

Reply:
@aviramha suggested only including the x86 and the shim, i dont know if there are any situations where macos would take arm64 and not arm64e?

Reply:
I've tested few cases and it always preferred arm64e, but I can't tell for sure ofc..

Reply:
it would be very easy to add it in if we ever needed to

Reply:
I guess my only concern is that it introduces a decision that macos can make without our input, which could lead to subtle bugs if we cant tell/ are incorrect about which binary is loaded from the fat binary

Reply:
I think there is no risk in including the arm64 library in the layer's fat binary, but there is cost, in binary size and compilation time, for including it twice in the mirrord binary. There is a certain risk in not including the arm64 library directly in the layer's fat binary, as there might be some scenarios we're unaware of, where an arm64e library cannot be loaded but an arm64 library can be. And if there aren't any such scenarios, they could be introduced by apple in the future.

Reply:
I took a bullet and dove into the best codebase:
https://github.com/apple-oss-distributions/dyld/blob/65bbeed63cec73f313b1d636e63f243964725a9d/common/MachOFile.cpp#L267
+
https://github.com/apple-oss-distributions/dyld/blob/65bbeed63cec73f313b1d636e63f243964725a9d/common/MachOFile.cpp#L315
Basically each "runtime" arch has an array of prioritized architectures it can load - you can see that arm64e always prefers arm64e - there's no "arm" runtime on macOS, only arm64e so we should be good!